File: C++/Exercícios Udemy/menor_de_tres.cpp
#include <iostream>
using namespace std;

void menor_de_tres(int a, int b, int c, int *menor);

int main() {
    int valor[3], menor, i;
    string classe[3] = {"Primeiro", "Segundo", "Terceiro"};
    for (i = 0; i < 3; i++) {
        cout << classe[i] << " valor: ";
        cin >> valor[i];
    }
    menor_de_tres(valor[0], valor[1], valor[2], &menor);
    cout << "MENOR = " << menor << endl;
    return 0;
}

void menor_de_tres(int a, int b, int c, int *menor) {
    *menor = a;
    if (b < *menor) *menor = b;
    if (c < *menor) *menor = c;
}

File: Java/Exercícios Udemy/menor_de_tres.java
import java.util.Locale;
import java.util.Scanner;

public class menor_de_tres {
    public static void main(String[] args) {
        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);
        int valor1, valor2, valor3, menor;
        System.out.println("Primeiro valor: ");
        valor1 = sc.nextInt();
        System.out.println("Segundo valor: ");
        valor2 = sc.nextInt();
        System.out.println("Terceiro valor: ");
        valor3 = sc.nextInt();
        menor = valor1;
        if (valor2 < menor) menor = valor2;
        if (valor3 < menor) menor = valor3;
        System.out.println("MENOR = " + menor);
        sc.close();
    }
}
